src/monattack.cpp

```diff
@@ -5812,3 +5812,92 @@ bool mattack::speaker( monster *z )
                    SNIPPET.random_from_category( "speaker_warning" ).value_or( translation() ) );
     return true;
 }
+
+bool mattack::dsa_drone_scan( monster *z )
+{
+    z->moves -= 100;
+    constexpr int scan_range = 30;
+    // Select a target: the avatar or a nearby NPC.  Must be visible and within scan range
+    Character *target = &get_player_character();
+    Character &you = get_player_character();
+    bool avatar_in_range = z->posz() == target->posz() && z->sees( target->pos() ) &&
```

Comment:
I understand this is a rare case, but that means that the drone won't be able to detect player from the roof or second floor?

Reply:
3DFoV is still so messed up that I'm tempted to keep this for now.  I guess the `sees()` should catch any wackiness like looking through the ceiling or floor.